internal/cluster/auth.go

```diff
@@ -0,0 +1,60 @@
+package cluster
```

Comment:
I'm not sure if this is the best place for this.

I wanted to avoid making it a standalone API that we would need to add to the giant list in the bminventory API, but it's not really all that closely related to cluster and we have plans to change the relationship between clusters and images in the future.

Any ideas for a better place to put this?

Reply:
I thought about putting it in `pkg/auth` but that's more about authenticating requests rather than issuing tokens.

The whole idea of using an asymmetric token was to reserve the ability to split the token generation from validation so it seems counter-productive to start them in the same package.